### ISSUE TITLE
Selected text highlight too hard to see on Macbook Pro. Need more color contrast.

### ISSUE THREAD

**Issue by rosscoleman** [jupyterlab/jupyterlab#13397]
https://github.com/jupyterlab/jupyterlab/blob/20bfb579e1b025ca0c7af321483185b971f82591/packages/theme-light-extension/style/variables.css#L329

When I select text in a JupyterLab code cell, I have a really difficult time seeing what is selected. That is because the highlight color is too close to white. It isn't so bad on an external (gaming) monitor, but it's really difficult to see on the built-in display of a 14 inch M1 Macbook Pro. 

`#d9d9d9` for the selected text in a code cell is just too close to the `#FFFFFF` page background color.

`--jp-editor-selected-background: #d9d9d9;`

**Comment by rosscoleman** [comment]
I am using JupyterLab 3.4.3. I think this may have been fixed in a later version, but I'm not an admin on our JupyterHub server, so I'm trying to figure that out.

**Comment by rosscoleman** [comment]
I found the problem. I forgot that I increased contrast after I had laser eye surgery. That makes grays look white, for some reason that isn't intuitive to me. When I set the contrast slider back to normal, I can see the highlights fine.

<img width="667" alt="image" src="https://user-images.githubusercontent.com/4631316/200734309-ac5700d0-f381-463c-9dbc-c630a5ada13b.png">

**Comment by jweill-aws** [comment]
The contrast between these two colors is 1.41:1, well below the WCAG requirements of 4.5:1 for normal text or 3:1 for large text. Source: https://webaim.org/resources/contrastchecker/?fcolor=D9D9D9&bcolor=FFFFFF . Thank you for filing this!

**Comment by jweill-aws** [comment]
The purplish background color in the first screen shot in [this comment](https://github.com/jupyterlab/jupyterlab/issues/13397#issuecomment-1308001002) is `#d7d4ee`, which has a contrast ratio of 1.44:1 with the `#ffffff` color used as general light background. Still too low of a contrast for WCAG guidelines.

https://webaim.org/resources/contrastchecker/?fcolor=D7D4EE&bcolor=FFFFFF

**Comment by krassowski** [comment]
Colours in syntax-highlighted editors are hard. Chance is that if you change a background colour, it will make another part of the line unreadable. See https://github.com/jupyterlab/jupyterlab/issues/10388 for an example. We could "free" some colours by using borders as suggested in https://github.com/jupyterlab/jupyterlab/issues/10388#issuecomment-861870577 (this is what chromium debugger and vscode do for active lines).

The WCAG requirements for text do **not** apply to contrast between non-text items. There is a separate Success Criterion [1.4.11 Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) which requires 3:1 and can be met by adding a border. Still, these guidelines are arbitrary and (for some definition) wrong (please see https://github.com/jupyter/accessibility/issues/65).

My opinion: we can either try adding a stronger contrast border, or close this issue since this was a user-side configuration problem. There should also be a high contrast theme for JupyterLab (also suggested in https://github.com/jupyterlab/jupyterlab/issues/2735#issuecomment-317503808), it appears that maybe one will be made part of https://github.com/Quansight-Labs/jupyterlab-accessible-themes (I don't know what is the status of that repo) and probably we should maintain it in the core, and in such a theme we could make conscious choices on which elements shall loose colour so that other can standout.

We **should** accept bugs when any of elements are not visible when applying the system-wide high contrast mode (available on Windows/Mac/Linux by default).
